### PR TITLE
Add return to proxyDirect to avoid double-connection.

### PR DESCRIPTION
**Pull request by 4wrxb**
This is to address issue #263 which seemed to only apply to cases where one of the gateway options was direct. Non-direct gateways worked properly since the return if err is nil was already present.

### CODE REVIEW

**Comment by moul** [comment]
Hi @4wrxb, the first part of this PR was cherry-picked in #285 and is now merged on master

For the second part, I currently have panic error when running it

As it seems to be related to another topic, can you update the title of this PR?

**Comment by 4wrxb** [comment]
TL;DR: I'm closing this pull request, only the first (now cherry picked) commit should have been there in the first place.

My apologies, I'm new to go so developing in the official branch (to keep compile working) and using other remotes is a new workflow for me.

This second commit is me hacking around for issue #268. As I mentioned there (and even included some comments with these changes), I don't fully understand the architecture I'm playing with (especially the part around line 190 which appears to only be used when listing hosts). The 2nd part is what allowed me to work around the issue described in 268.

**Comment by 4wrxb** [comment]
TL;DR: I'm closing this pull request, only the first (now cherry picked) commit should have been there in the first place.

My apologies, I'm new to go so developing in the official branch (to keep compile working) and using other remotes is a new workflow for me.

This second commit is me hacking around for issue #268. As I mentioned there (and even included some comments with these changes), I don't fully understand the architecture I'm playing with (especially the part around line 190 which appears to only be used when listing hosts). The 2nd part is what allowed me to work around the issue described in 268.
